src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import styles from './counter.module.scss';
+import { checkWindow } from '@/helpers/checkWindow';
+type Props = {
+  productId: number;
+  price: number;
+  setTotalPrice?: React.Dispatch<React.SetStateAction<number>>;
+  className?: string;
+  forceUpdate?: any;
+};
+
+export default function Counter({
+  productId,
+  price,
+  setTotalPrice,
+  className,
+  forceUpdate,
+}: Props) {
+  const [productCount, setProductCount] = useState(1);
+  const handlePlus = () => {
+    if (checkWindow()) {
+      setProductCount((prev) => prev + 1);
+      setTotalPrice && setTotalPrice((prev) => prev + price);
+      localStorage.setItem(productId.toString(), (productCount + 1).toString());
```

Comment:
You can create a separate custom hook for working with local storage, but it's just a suggestion.